twtxt/helper.py

```diff
@@ -85,3 +87,28 @@ def sort_and_truncate_tweets(tweets, direction, limit):
             return sorted(tweets)
     else:
         return []
+
+
+@contextlib.contextmanager
+def atomic_write(path, mode):
+    """
+        Simple contextmanager to safely write to a
+        file.
+
+        :param str filename: the path to write to
+        :param str mode: the mode to open the file
+    """
+    # generate random filepath in same location as given filename
+    tmpfilepath = os.path.join(os.path.dirname(path), ".{}".format(uuid.uuid4()))
```

Comment:
That was exactly the problem. For the rename it must be on the same partition - on my system the /tmp is on a different one - that's why I preferred the way with uuid. In addition the tempfile comes really handy with the contextmanagers and self-deletion however, I would have to disabled those.